src/components/UI/Button/index.js

```diff
@@ -1,17 +1,90 @@
 import PropTypes from "prop-types";
-import classNames from "classnames/bind";
-import styles from "./style.scss";
-
-const Button = ({ children, onClick, className, ...rest }) => {
-  const classes = classNames.bind(styles);
-  const getClassName = () => (className ? className.split(" ") : "");
-  const defaultClassName = classes("button", "pill", getClassName());
-  return (
-    <button className={defaultClassName} onClick={onClick} {...rest}>
-      {children}
-    </button>
-  );
-};
+import styled, { css } from "styled-components";
+import colors from "../../../styles/variables";
+
+const StyledButton = styled.button`
+  font-weight: bold;
+  cursor: pointer;
+  border: none;
+  padding: 1rem 2rem;
+  user-select: none;
+  text-decoration: none;
+  color: ${colors.white};
+  border: 1px solid ${colors.black};
+  background: ${colors.black};
+  min-width: 10rem;
+  margin: 8px;
+
+  ${props =>
+    props.type === "rounded" &&
+    css`
+      border-radius: 5px;
+    `};
+
+  ${props =>
+    props.type === "pill" &&
+    css`
+      border-radius: 50px;
+    `};
+
+  ${props =>
+    props.color === "orange" &&
```

Comment:
when we discussed about that, I suggested to create few components (like BlueButton) which you told me you didnt want to do, but in the end that's what we are doing here, for each color, we will need to define it before being usable.

I think it's better to give possibility to consumer/user to do what they want/need and eventually to provide a set of predefined components.

For instance emotion can do composition like explained => https://emotion.sh/docs/composition. It's probably also doable with styled-components (I didnt check the doc yet)

I suggest to 
- not have a prop color to color a component
- not have a specific case for outlining => either keep it always consistent by default, either create a new type (not sure this is a good idea, because what about outlined-pill ? I would go for option 1, keep it consistent)
- create a specific set of colored component, reusable, that anybody can take a look and create a new component based on

Reply:
I think the few component can be an issue later because the number of component you have to name can only scale as much as you can name them.. And you can get something like `BlueRedtextSolidborderButton`.

That versus

<Button
  color="red"
  bgColor="blue"
  border="solid"
/> 

The later seemed more manageable? 

I'm referencing material-ui (https://material-ui.com/components/buttons/) component it seemed like they are using one Button and then pass in properties like:

- variant
- color
- size

Reply:
I think I was not clear enough :)

Of course the idea of composing + specific set of component is not to create all of possible combination because it's no doable. The idea is to create a set of directly reusable components like https://getbootstrap.com/docs/4.0/components/buttons/ + the opportunity of developer to create specific set of components in a simple way for everybody.

Taking example of emotion + a component:

```javascript
import { css, jsx } from '@emotion/core
const base = css`
  background-color: blue;
  color: white;
`
export const Button = ({custom}) => <button css={[custom, base]}>Hello</button>

export const BlueOutlineButton = () => {
  const custom = css`
    background-color: green;
    color: dark;
    &:hover: {
      ...
    }
  `
  return <Button custom={custom}/>
}
```

With this you get simplicity of code + composition

Reply:
Hmmm this is doable and clean enough also. The logic is similar to composing inline styles. @thefiend got any concerns?

Reply:
This would get really big if we want to have all the combinations for the button of
Colours: Red, Blue, Green, Orange
Type: Rounded, Pill
Outline: Outline

It is probably better to use Conditional Class Names to allow the different combinations to be made through props passed into the button. Emotion now supports it too (https://emotion.sh/docs/emotion#cx), similar to classnames package.